docs/pages/docs/routing/middleware.mdx

```diff
@@ -549,73 +549,43 @@ export const config = {
 
 ### Example: Integrating with Supabase Authentication
 
-This example is mostly merging the next-intl middleware with the Supabase one. For more information on the Supabase middleware, check [this](https://supabase.com/docs/guides/auth/server-side/nextjs?router=app) guide from their website.
+In order to use Supabase Authentication with `next-intl`, you need to combine the Supabase middleware with the one from `next-intl`.
 
-First, we're going to set up two required environment variables in a `.env.local` file. The `NEXT_PUBLIC_SUPABASE_URL` and `NEXT_PUBLIC_SUPABASE_ANON_KEY` are required to create the `supabase` object.
-
-```txt filename=".env.local"
-NEXT_PUBLIC_SUPABASE_URL=your_supabase_project_url
-NEXT_PUBLIC_SUPABASE_ANON_KEY=your_supabase_anon_key
-```
-
-[`@supabase/ssr`](https://supabase.com/docs/guides/auth/server-side/creating-a-client) exports a `createServerClient` function. First we're going to create a function that invokes the `createServerClient` function and returns the `supabase` object.
-
-Finally, we're going to create a middleware file that uses the `createClient` function that we created and merges it with the `next-intl` middleware.
+You can do so by following the [setup guide from Supabase](https://supabase.com/docs/guides/auth/server-side/nextjs?router=app) and adapting the middleware as follows:
 
 ```tsx filename="middleware.ts"
 import {type NextRequest} from 'next/server';
 import {createServerClient, type CookieOptions} from '@supabase/ssr';
 import createIntlMiddleware from 'next-intl/middleware';
 
+const handleI18nRouting = createIntlMiddleware({
```

Comment:
Why put this function outside the middleware, if is not used anywhere else? Just curious

Reply:
I'd say it's really fine in either place. Since the creation of the middleware doesn't depend on any request-based information, I figured we can move it outside. But inside is absolutely ok too!